Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -219,6 +224,189 @@ noncomputable instance levyProkhorovDist_pseudoMetricSpace_probabilityMeasure :
+
+/-- Assuming `levyProkhorovEDist μ ν < ε`, we can bound `∫ f ∂μ` in terms of
+`∫ t in (0, ‖f‖], ν (thickening ε {x | f(x) ≥ t}) dt` and `‖f‖`. -/
+lemma BoundedContinuousFunction.integral_le_of_levyProkhorovEDist_lt (μ ν : Measure Ω)
+    [IsFiniteMeasure μ] [IsFiniteMeasure ν] {ε : ℝ} (ε_pos : 0 < ε)
+    (hμν : levyProkhorovEDist μ ν < ENNReal.ofReal ε) (f : Ω →ᵇ ℝ) (f_nn : 0 ≤ᵐ[μ] f) :
+    ∫ ω, f ω ∂μ
+      ≤ (∫ t in Ioc 0 ‖f‖, ENNReal.toReal (ν (thickening ε {a | t ≤ f a}))) + ε * ‖f‖ := by
+  rw [BoundedContinuousFunction.integral_eq_integral_meas_le f μ f_nn]
+  have key : (fun (t : ℝ) ↦ ENNReal.toReal (μ {a | t ≤ f a}))
+              ≤ (fun (t : ℝ) ↦ ENNReal.toReal (ν (thickening ε {a | t ≤ f a})) + ε) := by
+    intro t
+    convert (ENNReal.toReal_le_toReal (measure_ne_top _ _) ?_).mpr
+            <| left_measure_le_of_levyProkhorovEDist_lt hμν (B := {a | t ≤ f a})
+                (f.continuous.measurable measurableSet_Ici)
+    · rw [ENNReal.toReal_add (measure_ne_top ν _) ofReal_ne_top, ENNReal.toReal_ofReal ε_pos.le]
+    · exact ENNReal.add_ne_top.mpr ⟨measure_ne_top ν _, ofReal_ne_top⟩
+  have intble₁ : IntegrableOn (fun t ↦ ENNReal.toReal (μ {a | t ≤ f a})) (Ioc 0 ‖f‖) := by
+    apply Measure.integrableOn_of_bounded (M := ENNReal.toReal (μ univ)) measure_Ioc_lt_top
+    · apply (Measurable.ennreal_toReal (Antitone.measurable ?_)).aestronglyMeasurable
+      exact fun _ _ hst ↦ measure_mono (fun _ h ↦ hst.trans h)
+    · apply eventually_of_forall <| fun t ↦ ?_
+      simp only [Real.norm_eq_abs, abs_toReal]
+      exact (ENNReal.toReal_le_toReal (measure_ne_top _ _) (measure_ne_top _ _)).mpr
+            <| measure_mono (subset_univ _)
+  have intble₂ : IntegrableOn
+                  (fun t ↦ ENNReal.toReal (ν (thickening ε {a | t ≤ f a}))) (Ioc 0 ‖f‖) := by
+    apply Measure.integrableOn_of_bounded (M := ENNReal.toReal (ν univ)) measure_Ioc_lt_top
+    · apply (Measurable.ennreal_toReal (Antitone.measurable ?_)).aestronglyMeasurable
+      exact fun _ _ hst ↦ measure_mono <| thickening_subset_of_subset ε (fun _ h ↦ hst.trans h)
+    · apply eventually_of_forall <| fun t ↦ ?_
+      simp only [Real.norm_eq_abs, abs_toReal]
+      exact (ENNReal.toReal_le_toReal (measure_ne_top _ _) (measure_ne_top _ _)).mpr
+            <| measure_mono (subset_univ _)
+  apply le_trans (set_integral_mono (s := Ioc 0 ‖f‖) ?_ ?_ key)
+  rw [integral_add]
+  · apply add_le_add rfl.le
+    simp only [integral_const, MeasurableSet.univ, Measure.restrict_apply, univ_inter,
+                Real.volume_Ioc, sub_zero, norm_nonneg, toReal_ofReal, smul_eq_mul,
+                (mul_comm _ ε).le]
+  · exact intble₂
+  · exact integrable_const ε
+  · exact intble₁
+  · exact intble₂.add <| integrable_const ε
+
+/-- A monotone decreasing convergence lemma for integrals of measures of thickenings:
+`∫ t in (0, ‖f‖], μ (thickening ε {x | f(x) ≥ t}) dt` tends to
+`∫ t in (0, ‖f‖], μ {x | f(x) ≥ t} dt` as `ε → 0` along a sequence. -/
```

Comment:
Why do you take it along a subsequence, instead of along the filter of positive epsilons? With `tendsto_integral_filter_of_dominated_convergence`, your proof should generalize right away.

Reply:
Thank you, this is obviously an improvement.

The proof of `continuous_levyProkhorov_to_probabilityMeasure` still goes via sequential continuity. But since that is only the proof, not the statement, maybe that is not too bad.

In principle there should be a proof with filters, which should in fact be cleaner than my sequence proof. But at least for me, adding the epsilon of room with `exists_seq_strictAnti_tendsto` was easier than cleanly combining the neighborhood filter `𝓝 ν` in probability measures with an extra `𝓝[>] 0` filter... With sequences, I just took a sequence of probability measures and a sequence of positive epsilons tending to zero, and paired them up, because both were indexed by natural numbers. I'm sure there exists a clean and clever filter magic for the same, but perhaps the sequence proof is acceptable? (Again, here it is only the proof, not the statement, that is affected by my clumsiness.)